### DISCUSSION TITLE
Color picker/ theme editor

### DISCUSSION THREAD

**Discussion by B-Kramer** [home-assistant/frontend#3886] [Other feature requests]
While editing themes in the chrome dev console is fun, now that lovelace is default a new UI-driven theme editor with color picker (and maybe even suggested themes based on a color (https://coolors.co/app)) would be awesome and allow users to make awesome lovelace UI's.

It would also be a good exercise to nail down which themeing variables are going to be shown/supported.

maybe with: http://jscolor.com/examples/
or this: https://github.com/narsenico/a-color-picker/

**Comment by iantrich** [comment]
cc @thomasloven you mentioned doing something like this

**Comment by thomasloven** [comment]
Well, I have this: https://github.com/thomasloven/lovelace-theme-maker

There's no color picker yet, and I'm planning to add a list of common variables too.

**Comment by bramkragten** [comment]
The theming vars (yes, breaking changes...) should be cleaned up before we would start on an editor

**Comment by zsarnett** [comment]
https://github.com/home-assistant/frontend/pull/5224

**Comment by zsarnett** [comment]
Mockup of a per card theme editor. Though I think that most of these options should be based from a main theme. And then the theme tab would be able to edit specific values to the card. Sensor Graph color, Gauge Fill color, etc. But I can see an argument either way

(as you can see this is just a visual mockup. No working inputs)

![e26a7b217c6fbd634c242bada6748c82](https://user-images.githubusercontent.com/18730868/76804248-3177dd00-67b2-11ea-8004-867f46d88859.gif)

**Comment by KTibow** [comment]
Progress: We can set primary color and accent color to custom variables.

**Comment by KTibow** [comment]
Consider closing?

**Comment by zsarnett** [comment]
No Theme editor would be something different IMO

**Comment by zsarnett** [comment]
And I do still think we should add a theme Variable to the Sensor Graph

**Comment by KTibow** [comment]
Should this be moved to Discussions?
